sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuCartesianProductExec.scala

```diff
@@ -225,7 +229,10 @@ case class GpuCartesianProductExec(
         ret
       }
 
-      val maxRowCount = 100000000
+      // Hash aggregate explodes the rows out, so if we go too large
+      // it can blow up. The size of a Long is 8 bytes so we just go with
+      // that as our estimate, no nulls.
+      val maxRowCount = targetSizeBytes/8
```

Comment:
Nit: This should be Math.max(targetSizeBytes / 8, 1) or the loop below is going to do something bad if maxRowCount ever was zero.  Should not happen in practice just a theoretical thing.

Reply:
Wonder if it's more likely this could go the other way, and targetSizeBytes / 8 is more than MAX_INT rows.  Then we'd want to `Math.min(Int.maxValue, ...)` the result to make sure the truncation to int is safe.

Reply:
I suspect that there are lots of issues all over the code with batch sizes that even come close to `Int.maxValue`.  The reason for this limit is because down stream in hash aggregate it materializes a column and if we have `Int.maxValue` entries it blows up trying to turn a scalar into a column.  We also quickly run out of memory for large batches if I make the size a little smaller.  I'm not sure it is worth spending too much time on this now, but if you want me to "fix" this one case to make it always work I can.